### PR TITLE
Add organization type

### LINKED ISSUE #1187: Add Organization Type to Org Settings
We would like to know the type of setting for organizations for 2 reasons:

1. It allows us to do more analyses through Metabase (currently we categorize them by hand as needed) which we report on weekly and use to understand our user base.

2. It allows us to build setting specific features. For example showing school IDs to K12 schools and universities, or offering NHSN uploads to nursing homes.

- The Organization Type field will go on the Organization settings page. It should be set up as a part of the "Create Organization" workflow. 
- It should be a required field for creating an organization

Possible Answers:
- K-12 School
- College/University
- Correctional Facility
- Airport/Transit Station
- Homeless Shelter
- FQHC
- Primary Care / Mental Health Outpatient
- Assisted Living Facility
- Hospital or Clinic
- Urgent Care
- Nursing Home
- Substance Abuse Treatment Center
- Hospice
- Pharmacy
- Other

**TODO**

- [x] Add Organization Type field to Organization Settings page & App database
- [x] Update the Create Organization workflow to add Organization Settings as a required field
- [ ] Sent this data to ReportStream as "Site_of-Care" (see ticket #466)
- [ ] ensure this is set in the account-request/ form and that the signup form still works in a cloud env. Including that the data is still correctly sent to the test Dynamics instance

### PR DESCRIPTION
**Pull request by nickscialli-usds**
## Related Issue or Background Info

- Closes #1187

## Changes Proposed

- Adds `organization-type` as a column to organizations table
- ~~Reports as `Site_of_care` to ReportStream~~ (this part of the work has been moved to a separate ticket)
- Adds dropdown to change type in "Manage Organization" tab
- Handles registration from the static site

## Additional Information

- Needs to be deployed alongside https://github.com/CDCgov/prime-simplereport-site/pull/180

## Screenshots / Demos

organization dropdown

### UI
- [X] Any changes to the UI/UX are approved by design 
- [N/A] Any new or updated content (e.g. error messages) are approved by design 

### Testing
- [X] Includes a summary of what a code reviewer should verify

### Changes are Backwards Compatible
- [x] Database changes are submitted as a separate PR
  - [N/A] Any new tables that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
  - [N/A] Any changes to tables that have custom no-PHI views are accompanied by changes to those views
        (including re-granting permission to the no-PHI user if need be)
  - [N/A] Liquibase rollback has been tested locally using `./gradlew liquibaseRollbackSQL` or `liquibaseRollback`
- [X] GraphQL schema changes are backward compatible with older version of the front-end

### Security
- [N/A] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [N/A] Any dependencies introduced have been vetted and discussed

## Cloud
- [N/A] DevOps team has been notified if PR requires ops support
- [X] If there are changes that cannot be tested locally, this has been deployed to our Azure `test` environment for verification

### CODE REVIEW

File: backend/src/main/java/gov/cdc/usds/simplereport/api/model/TestEventExport.java [hunk 1]
```diff
@@ -476,4 +478,10 @@ public String getOrderTestDate() {
     // order_test_date = test_date for antigen testing
     return getTestDate();
   }
+
+  // Added 06/21/2021
+  @JsonProperty("Site_of_care")
```
Comment (nickscialli-usds) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/model/TestEventExport.java:483]: Do we have a good way to test this being sent to ReportStream?

**Review by hmyers-cms — APPROVED** [review]

File: backend/src/main/java/gov/cdc/usds/simplereport/db/model/Organization.java [hunk 2]
```diff
@@ -11,6 +11,9 @@
   @Column(nullable = false, unique = true)
   private String organizationName;
 
+  @Column(nullable = false)
```
Comment (nullflux) [inline backend/src/main/java/gov/cdc/usds/simplereport/db/model/Organization.java:14]: **question:** I noticed that `not null` is _not_ set on this field in the migration. Just curious: are we planning to backfill the data and then enforce this requirement there as well?

File: backend/src/main/resources/graphql/main.graphqls [hunk 3]
```diff
@@ -456,4 +457,6 @@ type Mutation {
   ): String @requiredPermissions(allOf: ["UPDATE_TEST"])
   sendPatientLinkSms(internalId: String!): String
     @requiredPermissions(allOf: ["UPDATE_TEST"])
+  updateOrganization(type: String!): String
```
Comment (nullflux) [inline backend/src/main/resources/graphql/main.graphqls:460]: **thought (non-blocking):** Unless I'm reading this wrong, I think this is a breaking change for an out of date UI

That said, I think it's a desirable one bc we don't want non-admins to be able to edit their own org name anymore

👍

File: backend/src/main/java/gov/cdc/usds/simplereport/service/model/crm/AccountRequestDynamicsData.java [hunk 4]
```diff
@@ -30,11 +30,8 @@ public AccountRequestDynamicsData(final AccountRequest accountRequest) {
     // variableMap.put("bah_state_text", accountRequest.getState()); // need mapping to dynamics
     variableMap.put("bah_zipcode_text", accountRequest.getZip());
     variableMap.put("bah_county_text", accountRequest.getCounty());
-    variableMap.put(
-        "bah_testingsitetype_code",
-        DynamicsValueMapping.convertToCode(Prefix.TST, accountRequest.getFacilityType()));
-    variableMap.put("bah_othertestingsite_text", accountRequest.getFacilityTypeOther());
     variableMap.put("bah_organizationname_text", accountRequest.getOrganizationName());
+    variableMap.put("bah_organizationtype_text", accountRequest.getOrganizationType());
```
Comment (nullflux) [inline backend/src/main/java/gov/cdc/usds/simplereport/service/model/crm/AccountRequestDynamicsData.java:34]: **praise:** Thanks for figuring out the Dynamics bit!

File: frontend/src/app/Settings/ManageOrganizationContainer.tsx [hunk 5]
```diff
@@ -60,30 +71,35 @@ const ManageOrganizationContainer: any = () => {
...
+      : () => setOrganization({ variables: { type } });
+
+    const alertProps: ComponentProps<typeof Alert> = {
+      type: "success",
+      title: "Updated Organization",
+      body: "The settings for the organization have been updated",
+    };
+
+    try {
+      await mutation();
```
Comment (nullflux) [inline frontend/src/app/Settings/ManageOrganizationContainer.tsx:89]: **praise:** Love this anonymous function + ternary combo

**Review by nullflux — APPROVED** [review]
LGTM :) 

Please bear w/ the awkward syntax, I'm trying out [conventional: comments](https://conventionalcomments.org/)

File: backend/src/main/java/gov/cdc/usds/simplereport/api/model/accountrequest/AccountRequest.java [hunk 6]
```diff
@@ -26,9 +26,8 @@
   @JsonProperty @NotNull private String state;
   @JsonProperty @NotNull private String zip;
   @JsonProperty @NotNull private String county;
-  @JsonProperty @NotNull private String facilityType;
-  @JsonProperty private String facilityTypeOther;
```
Comment (bmaden-usds) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/model/accountrequest/AccountRequest.java:30]: I would consider leaving these two fields so requests from the old UI can still be handled during the transition and removing them later.  Probability of that happening is probably small, so alternatively, we should just carefully watch for errors on `/account-request` for a little while after deploying.

**Review by bmaden-usds — CHANGES_REQUESTED** [review]
Looks good!  Needs just a couple of small changes to make sure the request to Dynamics gets accepted.

File: backend/src/main/java/gov/cdc/usds/simplereport/service/model/crm/AccountRequestDynamicsData.java (hunk 4 above)
Comment (nickscialli-usds) [reply backend/src/main/java/gov/cdc/usds/simplereport/service/model/crm/AccountRequestDynamicsData.java:34]: Do we have any control over the Dynamics fields at this point? Setting the main select to "Other" always and then specifying the org type in the "Other Testing Site Type" field feels a bit like a weird workaround.

File: frontend/src/app/Settings/ManageOrganizationContainer.tsx (hunk 5 above)
Comment (nickscialli-usds) [reply frontend/src/app/Settings/ManageOrganizationContainer.tsx:89]: ty!

**Review by bmaden-usds — APPROVED** [review]

**Pull request merged**